Fixing issue with cleaning product cache for all product after update blog post.
Now only cache for related products will be cleaned.

--- a/Model/Post.php
+++ b/Model/Post.php
@@ -581,8 +581,11 @@ class Post extends AbstractModel implements PostInterface, IdentityInterface
         if ($this->_appState->getAreaCode() == \Magento\Framework\App\Area::AREA_FRONTEND) {
             $identities[] = self::CACHE_TAG;
         }
-        if (is_array($this->getRelatedProductIds()) && count($this->getRelatedProductIds())) {
-            $identities[] = CatalogProduct::CACHE_TAG;
+        $relatedProductIds = $this->getRelatedProductIds();
+        if (is_array($relatedProductIds) && count($relatedProductIds)) {
+            foreach ($relatedProductIds as $relatedProductId) {
+                $identities[] = \Magento\Catalog\Model\Product::CACHE_TAG . '_' . $relatedProductId;
+            }
         }
         return $identities;
     }
